File: framework/scala/src/main/scala/soda/scala/web/Logger.scala
package soda.scala.web

import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

object Logger {

  def info(message: String): Unit = {
    log("INFO", message)
  }

  def error(message: String): Unit = {
    log("ERROR", message)
  }

  def exception(message: String, ex: Throwable): Unit = {
    error(s"$message\n${Utils.toString(ex)}")
  }

  private def formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")

  def log(level: String, message: String): Unit = {
    val dt = LocalDateTime.now().format(formatter)
    val threadName = Thread.currentThread().getName
    println(s"$dt [$threadName] [$level] $message")
  }

}

File: framework/kotlin/src/main/kotlin/soda/kotlin/web/Logger.kt
package soda.kotlin.web

import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

internal object Logger {

    private val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")

    fun log(level: String, message: String) {
        val dt = LocalDateTime.now().format(formatter)
        val threadName = Thread.currentThread().name
        System.err.println("$dt [$threadName] [$level] $message")
    }

    fun info(message: String) = log("INFO", message)

    fun error(message: String) = log("ERROR", message)

    fun exception(message: String, ex: Throwable) {
        error("$message\n${WebUtils.toString(ex)}")
    }

}